app/src/main/java/com/ftw/hometerview/adapter/RecyclerAdapter.kt

```diff
@@ -0,0 +1,59 @@
+package com.ftw.hometerview.adapter
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.databinding.DataBindingUtil
+import androidx.databinding.ViewDataBinding
+import androidx.recyclerview.widget.ListAdapter
+import androidx.recyclerview.widget.RecyclerView
+
+class DataBindingRecyclerAdapter : ListAdapter<RecyclerItem, BindingViewHolder>(
+    DiffCallback()
+) {
+    override fun getItemViewType(position: Int): Int {
+        return getItem(position).layoutId
+    }
+
+    override fun onCreateViewHolder(
+        parent: ViewGroup,
+        viewType: Int
+    ): BindingViewHolder {
+        val inflater = LayoutInflater.from(parent.context)
+        val binding: ViewDataBinding = DataBindingUtil.inflate(inflater, viewType, parent, false)
+        return BindingViewHolder(binding)
+    }
+
+    override fun onBindViewHolder(
+        holder: BindingViewHolder,
+        position: Int
+    ) {
+        holder.run {
+            getItem(position).bind(binding)
+            if (binding.hasPendingBindings()) {
+                binding.executePendingBindings()
+            }
+        }
+    }
+}
+
+class BindingViewHolder(
+    val binding: ViewDataBinding
+) : RecyclerView.ViewHolder(binding.root)
+
+private fun RecyclerItem.bind(binding: ViewDataBinding) {
+    val isVariableFound = binding.setVariable(variableId, data)
+    if (isVariableFound.not()) {
+        throw IllegalStateException(
+            buildErrorMessage(variableId, binding)
+        )
+    }
+}
```

Comment:
`onBindViewHolder()` 에서는 holder 의 bind() 함수를 호출할 것이라고 예상하는데, 여기에서는 그런 로직이 아니어서 좀 혼란스러워 보이는 것 같아. `run` 을 사용했을 때의 장점보다 가독성이 떨어지기도 하고... 

아래처럼 하면 어때?
```suggestion
    override fun onBindViewHolder(holder: BindingViewHolder, position: Int) {
        holder.bind(getItem(position))
    }
}

class BindingViewHolder(
    private val binding: ViewDataBinding
) : RecyclerView.ViewHolder(binding.root) {

    fun bind(item: RecyclerItem) {
        val isVariableFound = binding.setVariable(item.variableId, item.data)
        if (isVariableFound.not()) {
            throw IllegalStateException(
                buildErrorMessage(item.variableId, binding)
            )
        }

        if (binding.hasPendingBindings()) {
            binding.executePendingBindings()
        }
    }
}
```

Reply:
아 onBindViewHolder는 오로지 bind함에 의미가 있고, bind에서 유효성을 검사하는데에 집중한다는 의미이면 확실히 맥락도 맞고 가독성도 더 좋을 것 같아요! 수정하겠습니다.